### DISCUSSION TITLE
When using browser history, the x-model attribute doesn't keep the state in sync with the input element.

### DISCUSSION THREAD

**Discussion by emil-nasso** [alpinejs/alpine#2505] [7. Bugs (archived)]
I have a page with a search-bar that is handled using alpine.js. If I write something in the input-field, navigate away from the page and then get back to the page using the "back"-feature in my browser, I will still see my old input in the input field by the components state doesn't reflect that.

Here is a simplified reproduction of this problem:
https://jsfiddle.net/ynju46dg/14/

It seems to be working this way in chrome but firefox and safari doesn't repopulate the form when I use the back-feature.

Is this possibly a bug in alpine, or is it chrome that does something weird?

**Comment by danddanddand** [comment]
I think you might be on to something. In my application we have component that occasionally submits with empty data we can't replicate in development and I wonder if it's because user is navigating away, using back button and submitting form but state is empty because of this. Very interesting.

**Comment by SimoTod** [comment]
It's not strictly Alpine, it's the Back/Forward cache implementation in chrome that fills the input element with the previous value but, since it doesn't trigger an input event, the model is not aware of it.
It's definitely annoying though. In Alpine v3, as a workaround, you can add `@pageshow.window="$el.value = data"` to your input fields.

P.s. any reason why you used v2 for your demo? The code above won't work with Alpine v2 and you'll need to add an `x-ref` attribute for each input element and use `$refs...` instead of `$el`

**Reply by lukebouch** [reply]
@SimoTod I'm having the same problem and I tried adding `@pageshow.window="$el.value = data"` to each form input field but it does not seem to be doing anything.

**Reply by 7ammer** [reply]
> @SimoTod I'm having the same problem and I tried adding `@pageshow.window="$el.value = data"` to each form input field but it does not seem to be doing anything.

@lukebouch, try `@pageshow.window="data = $el.value"` instead. The workaround is the wrong way round :)

**Reply by lukebouch** [reply]
@7ammer Thanks!

**Comment by 7ammer** [comment]
This took me quite a while to figure out! Cheers for the workaround!

**Comment by alexandre-k** [comment]
Hello,
Another workaround would be to set the attribute `autocomplete=off` as an input field. Tested on Chrome. The other `autocomplete="nope"` is for Firefox.
Example:
```
<select
    x-model="myVariable"
    autocomplete="off"
    autocomplete="nope"
    >
    ...
</select>
```
